src/commands/synthetics/utils.ts

```diff
@@ -573,9 +577,18 @@ export const getTestsToTrigger = async (
     )
   )
 
+  // Keep track of uploaded applications to avoid uploading them twice.
```

Comment:
How all of these changes are related to the added flag?
Why remove the `&& overridenConfig` here?